includes/class-indieauth-token-endpoint.php

```diff
@@ -192,56 +228,89 @@ public function post( $request ) {
 		return new WP_OAuth_Response( 'invalid_request', __( 'Invalid Request', 'indieauth' ), 400 );
 	}
 
+
+	// Refresh Token Grant Type.
+	public function refresh_token( $params ) {
+
+		$diff = array_diff( array( 'refresh_token' ), array_keys( $params ) );
+		if ( ! empty( $diff ) ) {
+			return new WP_OAuth_Response( 'invalid_request', __( 'The request is missing one or more required parameters', 'indieauth' ), 400 );
+		}
+		$refresh = $this->refresh_tokens->get( $params['refresh_token'] );
+		if ( ! $refresh ) {
+			return new WP_OAuth_Response( 'invalid_grant', __( 'Invalid Token', 'indieauth' ), 400 );
+		}
+
+		// Destroy the refresh token.
+		$this->refresh_tokens->destroy( $params['refresh_token'] );
+
+		if ( array_key_exists( 'client_id', $params ) && ( $refresh['client_id'] !== $params['client_id'] ) ) {
```

Comment:
I think this isn't required - it's something I've mis-implemented on my side (confusing the request with how the credentialed clients work)

Reply:
Well, the client id is optional, so I debated not checking.